README.Rmd

```diff
@@ -0,0 +1,102 @@
+---
+output: github_document
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>",
+  fig.path = "man/figures/README-",
+  out.width = "100%"
+)
+```
+
+# Calculate the final size of an epidemic <img src="man/figures/logo.png" align="right" width="120"/>
+
+<!-- badges: start -->
+[![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
+[![R-CMD-check](https://github.com/epiverse-trace/finalsize/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/epiverse-trace/finalsize/actions/workflows/R-CMD-check.yaml)
+[![Codecov test coverage](https://codecov.io/gh/epiverse-trace/finalsize/branch/main/graph/badge.svg)](https://app.codecov.io/gh/epiverse-trace/finalsize?branch=main)
+<!-- badges: end -->
+
+_finalsize_ provides quick back-of-the-envelope calculations for the final size of an epidemic outbreak in a population with demographic variation in contact patterns, and variation within and between age groups in their susceptibility to disease.
+
+_finalsize_ can help provide rough estimates of the effectiveness of pharmaceutical interventions in the form of immunisation programmes, or the effect of naturally acquired immunity through previous infection (see the vignette).
+
+_finalsize_ relies on [Eigen](https://eigen.tuxfamily.org/) via [RcppEigen](https://github.com/RcppCore/RcppEigen) for fast matrix algebra, and is developed at the [Centre for the Mathematical Modelling of Infectious Diseases](https://www.lshtm.ac.uk/research/centres/centre-mathematical-modelling-infectious-diseases) at the London School of Hygiene and Tropical Medicine as part of the [Epiverse Initiative](https://data.org/initiatives/epiverse/).
+
+## Installation
+
+You can install the development version of finalsize from [GitHub](https://github.com/) with:
+
+``` r
+# install.packages("devtools")
+devtools::install_github("epiverse-trace/finalsize")
+
+# load finalsize
+library(finalsize)
+```
+
+## Quick start
+
+_finalsize_ provides the single function `final_size()`, to calculate the final size of an epidemic.
+
+Here, an example using social contact data from the `socialmixr` package investigates the final size of an epidemic when the disease has an R<sub>0</sub> of 2.0, and given three age groups of interest --- 0-19, 20-39 and 40+.
+The under-20 age group is assumed to be fully susceptible to the disease, whereas individuals aged over 20 are only half as susceptible.
+
+```r
+# Load socialmixr package
+# install.packages("socialmixr")
+library(socialmixr)
+
+# Load data from POLYMOD
+data(polymod)
+contact_data <- contact_matrix(
+  polymod, 
+  countries = "United Kingdom", 
+  age.limits = c(0,20,40)
+)
+
+# Define contact matrix (entry {ij} is contacts in group i reported by group j)
+contact_matrix <- t(contact_data$matrix)
+
+# Define population in each age group
+demography_vector <- contact_data$demography$population
```

Comment:
I tried running this example locally, but got an error. I think it stems from L64 as `contact_data` only has `$matrix` or `$participants` but not `$demography`. I re-installed `socialmixr` to check if it was due to having an outdated version but this issue persisted.

Reply:
My bad, that's due to the earlier `socialmixr::contact_matrix` argument `symmetric` being ommitted.

Reply:
Updated and checked with a local `devtools::build_readme` run